requests_futures/sessions.py

```diff
@@ -51,16 +51,19 @@ def __init__(self, executor=None, max_workers=2, session=None, *args,
         * If you provide both `executor` and `max_workers`, the latter is
           ignored and provided executor is used as is.
         """
+        _adapter_kwargs = {}
         super(FuturesSession, self).__init__(*args, **kwargs)
         self._owned_executor = executor is None
         if executor is None:
             executor = ThreadPoolExecutor(max_workers=max_workers)
             # set connection pool size equal to max_workers if needed
             if max_workers > DEFAULT_POOLSIZE:
-                adapter_kwargs = dict(pool_connections=max_workers,
-                                      pool_maxsize=max_workers)
-                self.mount('https://', HTTPAdapter(**adapter_kwargs))
-                self.mount('http://', HTTPAdapter(**adapter_kwargs))
+                _adapter_kwargs.update({'pool_connections': max_workers,
+                                       'pool_maxsize': max_workers})
+
+        _adapter_kwargs.update(adapter_kwargs or {})
```

Comment:
The `or {}` can likely be removed here and the default of `None` up above changed to `{}`.

Reply:
I generally try to avoid using Python's mutable types (`dict`, `list`, etc) as default arguments because of some of the oddities they bring. That said, those oddities might not pop up in this scenario but I'd still like to err on the side of caution. Will change if you feel adamantly.